src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2164,8 +2164,6 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                 if (operandComparedToNull != null)
                 {
-                    operandComparedToNull = SkipReferenceConversions(operandComparedToNull);
```

Comment:
>operandComparedToNull = SkipReferenceConversions(operandComparedToNull); [](start = 20, length = 72)

FYI @jcouv I found removing this conversion-stripping step did not affect anything.